src/k8s-extension/azext_k8s_extension/partner_extensions/OpenServiceMesh.py

```diff
@@ -93,3 +102,63 @@ def Update(self, extension, auto_upgrade_minor_version, release_train, version):
             release_train=release_train,
             version=version
         )
+
+    def Delete(self, client, resource_group_name, cluster_name, name, cluster_type):
+        pass
+
+
+def _validate_tested_distro(cmd, cluster_resource_group_name, cluster_name, extension_version):
+
+    field_unavailable_error = '\"testedDistros\" field unavailable for version {0} of microsoft.openservicemesh, ' \
+        'cannot determine if this Kubernetes distribution has been properly tested'.format(extension_version)
+
+    if version.parse(str(extension_version)) <= version.parse("0.8.3"):
+        logger.warning(field_unavailable_error)
+        return
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+    resources = cf_resources(cmd.cli_ctx, subscription_id)
+
+    cluster_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Kubernetes' \
+        '/connectedClusters/{2}'.format(subscription_id, cluster_resource_group_name, cluster_name)
+
+    resource = resources.get_by_id(cluster_resource_id, '2020-01-01-preview')
+    cluster_distro = resource.properties['distribution'].lower()
+
+    if cluster_distro == "general":
+        logger.warning('Unable to determine if distro has been tested for microsoft.openservicemesh, '
+                       'kubernetes distro: \"general\"')
+        return
+
+    tested_distros = _get_tested_distros(extension_version)
+
+    if tested_distros is None:
+        logger.warning(field_unavailable_error)
+    elif cluster_distro not in tested_distros.split():
+        logger.warning('Untested kubernetes distro for microsoft.openservicemesh, Kubernetes distro is %s',
+                       cluster_distro)
+
+
+def _get_tested_distros(chart_version):
+
+    try:
+        chart_arc = ChartBuilder({
+            "name": OpenServiceMesh.CHART_NAME,
+            "version": str(chart_version),
+            "source": {
+                "type": "repo",
+                "location": OpenServiceMesh.CHART_LOCATION
+            }
+        })
+    except VersionError:
+        raise InvalidArgumentValueError(
+            "Invalid version '{}' for microsoft.openservicemesh".format(chart_version)
+        )
+
+    values = chart_arc.get_values()
+    values_yaml = yaml.load(values.raw, Loader=yaml.FullLoader)
+
+    try:
+        return values_yaml['OpenServiceMesh']['testedDistros']
+    except KeyError:
+        return None

```

Comment:
If it's just for testing, could we move it to the test file `test_open_service_mesh.py`?

Reply:
This is not for testing, this is something that they want to alert users of in production